python/ray/tune/search/optuna/optuna_search.py

```diff
@@ -362,6 +372,15 @@ def __init__(
         self._sampler = sampler
         self._seed = seed
 
+        if storage:
+            assert isinstance(storage, BaseStorage), (
+                "The `storage` parameter in `OptunaSearcher` must be an instance "
+                "of `optuna.samplers.BaseStorage`."
+            )
+            self._storage = storage
+        else:
+            self._storage = ot.storages.InMemoryStorage()
```

Comment:
`InMemoryStorage` only exists in Optuna 4.x. Let's just pass in None, which works for both 3.x and 4.x:
* https://optuna.readthedocs.io/en/v3.6.1/reference/generated/optuna.create_study.html#optuna.create_study
* https://optuna.readthedocs.io/en/stable/reference/generated/optuna.create_study.html

Reply:
Good catch! Let me quickly fix it.